include/seqan3/alphabet/aminoacid/aa20.hpp

```diff
@@ -0,0 +1,412 @@
+    template <typename other_aa_type>
+    //!\cond
+        requires aminoacid_concept<other_aa_type>
+    //!\endcond
+    explicit constexpr operator other_aa_type() const noexcept
+    {
+        return detail::convert_through_char_representation<other_aa_type, std::decay_t<decltype(*this)>>[to_rank()];
+    }
+    //!\}
+
+    //!\name Comparison operators
+    //!\{
+    constexpr bool operator==(aa20 const & rhs) const noexcept
+    {
+        return _value == rhs._value;
+    }
+
+    constexpr bool operator!=(aa20 const & rhs) const noexcept
+    {
+        return _value != rhs._value;
+    }
+
+    constexpr bool operator<(aa20 const & rhs) const noexcept
+    {
+        return _value < rhs._value;
+    }
+
+    constexpr bool operator>(aa20 const & rhs) const noexcept
+    {
+        return _value > rhs._value;
+    }
+
+    constexpr bool operator<=(aa20 const & rhs) const noexcept
+    {
+        return _value <= rhs._value;
+    }
+
+    constexpr bool operator>=(aa20 const & rhs) const noexcept
+    {
+        return _value >= rhs._value;
+    }
+    //!\}
+
+    protected:
+    //!\privatesection
+    /*!\brief The internal type is a strictly typed enum.
+     *
+     * This is done to prevent aggregate initialization from numbers and/or chars.
+     * It is has the drawback that it also introduces a scope which in turn makes
+     * the static "letter values" members necessary.
+     */
+    enum struct internal_type : rank_type
+    {
+        A,
+        C,
+        D,
+        E,
+        F,
+        G,
+        H,
+        I,
+        K,
+        L,
+        M,
+        N,
+        P,
+        Q,
+        R,
+        S,
+        T,
+        V,
+        W,
+        Y,
+        B = D,
+        J = L,
+        O = L,
+        U = C,
+        X = S,
+        Z = E,
+        TERMINATOR = W,
+        UNKNOWN = S
+    };
+
+    //!\brief Value to char conversion table.
+    static constexpr char_type value_to_char[value_size]
+    {
+        'A',
+        'C',
+        'D',
+        'E',
+        'F',
+        'G',
+        'H',
+        'I',
+        'K',
+        'L',
+        'M',
+        'N',
+        'P',
+        'Q',
+        'R',
+        'S',
+        'T',
+        'V',
+        'W',
+        'Y',
+    };
+
+    //!\brief Char to value conversion table.
+    static constexpr std::array<internal_type, 256> char_to_value
+    {
+        [] () constexpr
+        {
+            using in_t = internal_type;
+            std::array<in_t, 256> ret{};
+
+            // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
+            for (auto & c : ret)
+                c = in_t::UNKNOWN;
+
+            ret['A'] = in_t::A; ret['a'] = in_t::A;
+            ret['B'] = in_t::D; ret['b'] = in_t::D; // Convert b (either D/N) to D, since D occurs more frequently. King, J. L. and T. H. Jukes. 1969. Non-Darwinian evolution. Science 164:788-798
```

Comment:
These comments are very helpful! Sadly, they cannot be seen in the User Documentation in this way. Consider doxygen comments.

Reply:
Line length